### DISCUSSION TITLE
Can't get the hosted version to connect to any of my printers or the simulator

### DISCUSSION THREAD

**Discussion by NameOfTheDragon** [mainsail-crew/mainsail#1961] [Help]
Hi Mainsail Crew. I can't seem to be able to connect to any of my printers anymore using the hosted version of Mainsail at my.mainsail.xyz. Nor, for that matter, can I get my printers to connect to each other, or to the virtual Klipper printer running in Docker. I used to be able to do this, so I'm not sure what changed. I just get a spinner that eventually times out.

I believe I have all the correcr CORS settings etc. - here is what I believe is the critical part:  

```yaml
[server]
host: 0.0.0.0
port: 7125
klippy_uds_address: /home/biqu/printer_data/comms/klippy.sock


[authorization]
cors_domains:
    *://my.mainsail.xyz
    *://*.local
trusted_clients:
    10.0.0.0/8
    127.0.0.0/8
    169.254.0.0/16
    172.16.0.0/12
    192.168.0.0/16
    FE80::/10
    ::1/128

[octoprint_compat]
```

So am I missing some configuration, or is there an issue with the code?

**Comment by meteyou** [comment]
hey @NameOfTheDragon,

pay attention, that you have to use `http://my.mainsail.xyz` instead of `https://my.mainsail.xyz`, because you don't have a certificate added in moonraker. pls check the url in your browser.

**Comment by NameOfTheDragon** [comment]
This is indeed the problem - however it is not obvious that it's happening unless you look really closely because MS Edge is doing the redirection without so much as a by-your-leave. I could not find any way to get to the insecure URL until I added an exception for the site in Edge's settings:
![image](https://github.com/user-attachments/assets/b1405f0f-19fa-4829-898e-e8bf42af64fd)

How would I add the certificate if I needed to? Which certificate?

**Comment by meteyou** [comment]
this is really very sad for the DIY domain. unfortunately all browsers are putting more and more obstacles in the way and so far i have not found a way that EVERY user can easily create certificates that are valid in his home network.

here you can find the settings and a short explanation of where to save the certificates in moonraker: https://moonraker.readthedocs.io/en/latest/configuration/#server

furthermore, i would recommend every user to use the locally installed Mainsail and not my.mainsail.xyz, as certain features do not work with the externally hosted web interface (also a browser limitation). may i ask why you want to use my.mainsail.xyz instead of the locally installed one?
